From 893bcebc5a6dd742258eb6d0f866838ecc01e8c5 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Sun, 13 Mar 2022 08:19:30 +0200
Subject: [PATCH] [SW-56501] gaudi2: config: increase page size

Increase page size to 64KB, to allow allocating large
physical contiguous memory from HBMs space for Linux,
to be used by arc core. This will allow using up to 64KB
contiguous memory.

Change-Id: Ice7cb79836e369f28b95d5b1cc20a0ae6fb6ad30
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/board/habanalabs/gaudi2/linux_asic.fragment b/board/habanalabs/gaudi2/linux_asic.fragment
index 434094a..1a3ecf3 100644
--- a/board/habanalabs/gaudi2/linux_asic.fragment
+++ b/board/habanalabs/gaudi2/linux_asic.fragment
@@ -28,3 +28,4 @@
 CONFIG_JFFS2_FS=y
 CONFIG_POSIX_MQUEUE=y
 CONFIG_I2C=n
+CONFIG_ARM64_64K_PAGES=y
diff --git a/board/habanalabs/gaudi2/linux_onix.fragment b/board/habanalabs/gaudi2/linux_onix.fragment
index dc1aa6e..bbd1851 100644
--- a/board/habanalabs/gaudi2/linux_onix.fragment
+++ b/board/habanalabs/gaudi2/linux_onix.fragment
@@ -29,3 +29,4 @@
 CONFIG_JFFS2_FS=y
 CONFIG_POSIX_MQUEUE=y
 CONFIG_I2C=n
+CONFIG_ARM64_64K_PAGES=y
diff --git a/board/habanalabs/gaudi2/linux_pldm.fragment b/board/habanalabs/gaudi2/linux_pldm.fragment
index 23a52b5..7e19790 100644
--- a/board/habanalabs/gaudi2/linux_pldm.fragment
+++ b/board/habanalabs/gaudi2/linux_pldm.fragment
@@ -4,3 +4,4 @@
 CONFIG_NR_CPUS=2
 CONFIG_HZ_12=y
 CONFIG_CMDLINE="earlycon console=ttyMPS0,1200n8 root=/dev/ram initcall_debug debug mem=160M"
+CONFIG_ARM64_64K_PAGES=y
